src/nvim/ex_docmd.c

```diff
@@ -2679,17 +2679,25 @@ set_one_cmd_context (
     p = cmd + 1;
   } else {
     p = cmd;
-    while (ASCII_ISALPHA(*p) || *p == '*')        /* Allow * wild card */
+    while (ASCII_ISALPHA(*p) || *p == '*') {  // Allow * wild card
       ++p;
-    /* check for non-alpha command */
-    if (p == cmd && vim_strchr((char_u *)"@*!=><&~#", *p) != NULL)
-      ++p;
-    /* for python 3.x: ":py3*" commands completion */
+    }
+    // a user command may contain digits
+    if (ASCII_ISUPPER(cmd[0])) {
+      while (ASCII_ISALNUM(*p) || *p == '*') {
+        ++p;
```

Comment:
Our style guide not allow us to use pre-increment as statement.

see: https://neovim.io/develop/style-guide.xml?showone=Postincrement_and_Postdecrement#Postincrement_and_Postdecrement